src/biotite/structure/io/pdbx/convert.py

```diff
@@ -211,97 +237,137 @@ def get_structure(pdbx_file, model=None, data_block=None, altloc="first",
         # Any field of the category would work here to get the length
         model_length = len(model_dict["group_PDB"])
         array = AtomArray(model_length)
-        
+
         _fill_annotations(array, model_dict, extra_fields, use_author_fields)
-        
+
         # Append exclusive stop
-        model_starts = np.append(
-            model_starts, [len(atom_site_dict["group_PDB"])]
-        )
+        model_starts = np.append(model_starts, [len(atom_site_dict["group_PDB"])])
         # Indexing starts at 0, but model number starts at 1
         model_index = model - 1
-        start, stop = model_starts[model_index], model_starts[model_index+1]
+        start, stop = model_starts[model_index], model_starts[model_index + 1]
         array.coord = np.zeros((model_length, 3), dtype=np.float32)
-        array.coord[:,0] = atom_site_dict["Cartn_x"][start : stop] \
-                           .astype(np.float32)
-        array.coord[:,1] = atom_site_dict["Cartn_y"][start : stop] \
-                           .astype(np.float32)
-        array.coord[:,2] = atom_site_dict["Cartn_z"][start : stop] \
-                           .astype(np.float32)
-        
+        array.coord[:, 0] = atom_site_dict["Cartn_x"][start:stop].astype(np.float32)
+        array.coord[:, 1] = atom_site_dict["Cartn_y"][start:stop].astype(np.float32)
+        array.coord[:, 2] = atom_site_dict["Cartn_z"][start:stop].astype(np.float32)
+
         array = _filter_altloc(array, model_dict, altloc)
-        
+
         array.box = _get_box(pdbx_file, data_block)
-        
+
         return array
-        
 
-def _fill_annotations(array, model_dict, extra_fields, use_author_fields):
-    prefix = "auth" if use_author_fields else "label"
-    array.set_annotation(
-        "chain_id", model_dict[f"{prefix}_asym_id"].astype("U4")
-    )
-    array.set_annotation(
-        "res_id", np.array(
-            [-1 if e in [".","?"] else int(e)
-             for e in model_dict[f"{prefix}_seq_id"]]
+
+def _fill_annotations(
+    array,
+    model_dict,
+    extra_fields,
+    use_author_fields,
+):
+    """Fill atom_site annotations in atom array or atom array stack.
+
+    Parameters
+    ----------
+    array : Union[AtomArrayStack, AtomArray]
+        Atom stack which will be annotated
+    model_dict : Dict[str, npt.NDArray[Any]]
+        ``atom_site`` dictionary with values for one model.
+    extra_fields : List[str]
+        Entry names, that are additionally added as annotation arrays.
+    use_author_fields : bool
+        Define if alternate fields prefixed with auth should be used instead of label
+        prefix
+    """
+
+    def get_or_fallback_from_dict(input_dict, key, fallback_key):
+        """
+        Return value related to key in input dict if it exists otherwise try to get the
+        value related to fallback key."""
+        return input_dict[key if key in input_dict else fallback_key]
```

Comment:
I think a warning should be raised here, if a fallback is used. Furthermore, in my opinion an `InvalidFileError` should be given, that state the missing *key*, if neither the key or the fallback work. This would enable the user to trace the origin of the error in the input file.

Reply:
Indeed it makes sense, is the message explicit enough ?